### DISCUSSION TITLE
Any way change profile settings?

### DISCUSSION THREAD

**Discussion by OliverTrust** [ValvePython/steam#326] [Q&A] [answered]
Avatar - no way,
nickname and status - change_status()
profile link - ?
real name - ?
country - ?
About myself - ?

And other settings (profile theme and etc)

**Comment by rossengeorgiev** [comment]
Not at the moment. There is an attempt at avatar upload in #290

**Comment by step-FLOOR-ring** [comment]
Are there any plans to get community settings changed in the near future? Wouldn't mind the ability to change custom profile URLs!

**Comment by rossengeorgiev** [comment, marked as answer]
It's just a single request.

```python
wa = WebAuth('<username>')
wa.cli_login()

resp = wa.session.post('https://steamcommunity.com/profiles/{:d}/edit/'.format(wa.steam_id),
                       files={'json': (None, 1)},  # forces multi-part encoding
                       data={
                         'sessionID': a.session_id,
                         'type': 'profileSave',
                         'personaName': 'PinkComodo',
                         'real_name': 'John Smith',
                         'customURL': '',
                         'summary': 'line1\nline2\nline3\n🔥🔥🔥',
                         'country': 'US',
                         'state': 'CA',
                         'city': 239,  # Fairfield
                         'hide_profile_awards': 1,
                       })

print(resp.json())
# {'success': 1, 'errmsg': ''}
```


https://steamcommunity.com//actions/QueryLocations/ - Gives you countries, and CC
https://steamcommunity.com//actions/QueryLocations/US/ - Gives you states
https://steamcommunity.com//actions/QueryLocations/US/CO - Gives you cities and their `cityid`

**Comment by OliverTrust** [comment]
@rossengeorgiev. awesome! Thank you! The only thing left is to change the avatar...

**Comment by rossengeorgiev** [comment]
Setting avatar: https://github.com/ValvePython/steam/pull/290/files#diff-c5dcf5183586df09601971f83d3f028443ed37da81bb01170cf1a6b0de0ea9caR36-R58

**Comment by OliverTrust** [comment]
Try this way, but can't upload avatar. All time `'success': False`
```python
avatar = open('1.png', 'rb')
wa = WebAuth(username='login')
wa.cli_login(password='password')
set_avatat_resp = wa.session.post('https://steamcommunity.com/actions/FileUploader/',
                                  timeout=15,
                                  data={
                                      'type': 'player_avatar_image',
                                       'sId': wa.steam_id,
                                       'sessionid': wa.session_id,
                                       'json': 1
                                   },
                                   files={'avatar': avatar}).json()

print(set_avatat_resp)
# {'success': False, 'images': [], 'hash': None, 'message': ''}
```

**Reply by rossengeorgiev** [reply]
You have to see what the browser sends exactly, and replicate the same request. Also, make sure the image is the correct size.

**Reply by rossengeorgiev** [reply]
Hi @OliverTrust, did you make your avatar upload work?
